### PR TITLE
Added findByContainerAndStatus

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-716

### Change description ###
Added findByContainerAndStatus to EnvelopeRepository

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=5285279&bid=17303756)

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepositoryTest.java [hunk 1]
```diff
@@ -126,6 +128,57 @@ public void should_get_1_incomplete_envelope_from_yesterday() {
...
+            envelope("Z", Status.COMPLETED, "container3")
+        );
+
+        // when
+        final List<Envelope> result = repo.findByContainerAndStatus(container1, Status.COMPLETED);
+
+        // then
+        assertThat(result)
+            .hasSize(2)
+            .extracting(this::getTupleFromEnvelope)
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepositoryTest.java:149]: could be  oneliner anyway: `envelope -> tuple(envelope.status, envelope.container)`

Comment (apereverzin) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepositoryTest.java]: fixed

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=5285279&bid=17303756)

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=5285279&bid=17303756)

**Pull request merged**